src/middlewares/guest_require.js

```diff
@@ -0,0 +1,14 @@
+import {
+    redirect,
+    Navigate
+  } from "react-router-dom";
+const GuestRequire = ({children})=>{
+    const user = localStorage.getItem('user')
+    if(user){
+        return <Navigate to={'/chat'} />
+    }else{
+        return children
+    }
+}
+
+export default GuestRequire
```

Comment:
its the same component as above, create Require component that get a route prop.